### PR TITLE
[9.0] remove JobDB's site mask

### PR DESCRIPTION
**Pull request by fstagni**
- [x] add wiki
- [x] mod DiracX

BEGINRELEASENOTES

*WMS
CHANGE: removed JobDB's SiteMask and Logging

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 1]
```diff
@@ -32,136 +32,106 @@ def initializeHandler(cls, svcInfoDict):
...
+        return S_OK()
 
     ##############################################################################
     types_getSiteMask = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getSiteMask(cls, siteState="Active"):
         """Get the site mask
-
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:52]: need a new line before the `:param` blocks.
this causes the errors in read the docs builds

File: docs/source/AdministratorGuide/Systems/ResourceStatus/install.rst [hunk 2]
```diff
@@ -112,13 +112,6 @@ If is there any SE to be modified, you can do it as follows::
 
 .. _activateRSS:
```
Comment (andresailer) [inline docs/source/AdministratorGuide/Systems/ResourceStatus/install.rst:113]: Drop this as well?

And RSS is active by default?

Comment (fstagni) [reply docs/source/AdministratorGuide/Systems/ResourceStatus/install.rst:113]: Yes, RSS is active by default.

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 3]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
 
         return S_OK()
 
-    types_setSiteMask = [list]
-
+    @deprecated("no-op RPC")
     def export_setSiteMask(self, siteList):
         """Set the site mask for matching. The mask is given in a form of Classad string.
-
         :param list siteList: site, status
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:38]:
```suggestion

        :param list siteList: site, status
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 4]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
 
     ##############################################################################
     types_getSiteMask = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getSiteMask(cls, siteState="Active"):
         """Get the site mask
-
         :param str siteState: site status
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:51]:
```suggestion

        :param str siteState: site status
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 5]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
+        return S_OK()
 
     types_getSiteMaskStatus = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getSiteMaskStatus(cls, sites=None):
         """Get the site mask of given site(s) with columns 'site' and 'status' only
-
         :param sites: list of sites or site
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:63]:
```suggestion

        :param sites: list of sites or site
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 6]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
 
     ##############################################################################
     types_getAllSiteMaskStatus = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getAllSiteMaskStatus(cls):
         """Get all the site parameters in the site mask
-
         :return: dict
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:77]:
```suggestion

        :return: dict
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 7]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
+        return S_OK()
 
     ##############################################################################
     types_banSite = [str]
 
+    @deprecated("no-op RPC")
     def export_banSite(self, site, comment="No comment"):
         """Ban the given site in the site mask
-
         :param str site: site
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:88]:
```suggestion

        :param str site: site
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 8]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
+        return S_OK()
 
     ##############################################################################
     types_allowSite = [str]
 
+    @deprecated("no-op RPC")
     def export_allowSite(self, site, comment="No comment"):
         """Allow the given site in the site mask
-
         :param str site: site
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:101]:
```suggestion

        :param str site: site
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 9]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
 
     ##############################################################################
     types_clearMask = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_clearMask(cls):
         """Clear up the entire site mask
-
         :return: S_OK()/S_ERROR()
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:115]:
```suggestion

        :return: S_OK()/S_ERROR()
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 10]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
 
     ##############################################################################
     types_getSiteMaskLogging = [[str, list]]
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getSiteMaskLogging(cls, sites):
         """Get the site mask logging history
-
         :param list sites: sites
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:127]:
```suggestion

        :param list sites: sites
```

File: src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py [hunk 11]
```diff
@@ -30,136 +31,106 @@ def initializeHandler(cls, svcInfoDict):
...
 
     ##############################################################################
     types_getSiteMaskSummary = []
 
     @classmethod
+    @deprecated("no-op RPC")
     def export_getSiteMaskSummary(cls):
         """Get the mask status for all the configured sites
-
         :return: S_OK(dict)/S_ERROR()
```
Comment (andresailer) [inline src/DIRAC/WorkloadManagementSystem/Service/WMSAdministratorHandler.py:140]:
```suggestion

        :return: S_OK(dict)/S_ERROR()
```

File: src/DIRAC/Interfaces/API/DiracAdmin.py [hunk 12]
```diff
@@ -157,34 +155,30 @@ def getSiteSection(self, site, printOutput=False):
...
-        result = self._checkSiteIsValid(site)
-        if not result["OK"]:
+        if not (result := self._checkSiteIsValid(site))["OK"]:
             return result
 
-        result = self.getSiteMask(status="Active")
-        if not result["OK"]:
+        if not (result := self.getSiteMask(status="Active"))["OK"]:
             return result
         siteMask = result["Value"]
```
Comment (chaen) [inline src/DIRAC/Interfaces/API/DiracAdmin.py:175]: `result` is the previous `result` here if you use this walrus operator no ?

File: src/DIRAC/DataManagementSystem/Client/test/Test_FTS3Objects.py [hunk 13]
```diff
@@ -181,6 +172,20 @@ def monkeypatchForAllTest(monkeypatch):
         lambda _self, _seName, _vo: S_OK(),
     )
 
+    def mock_init(self, useProxy=False, vo=None):
```
Comment (chaen) [inline src/DIRAC/DataManagementSystem/Client/test/Test_FTS3Objects.py:175]: Why is that necessary here and below now ?

File: src/DIRAC/Interfaces/API/DiracAdmin.py (hunk 12 above)
Comment (fstagni) [reply src/DIRAC/Interfaces/API/DiracAdmin.py:175]: Mmm no...

```py
if not (result := self._checkSiteIsValid(site))["OK"]:
    return result

if not (result := self.getSiteMask(status="Active"))["OK"]:
    return result
```

and

```py
result = self._checkSiteIsValid(site)
if not result["OK"]: 
    return result

result = self.getSiteMask(status="Active")
if not result["OK"]:
    return result
```

seem logically equivalent to me (and also according to chatGPT)

File: src/DIRAC/DataManagementSystem/Client/test/Test_FTS3Objects.py (hunk 13 above)
Comment (fstagni) [reply src/DIRAC/DataManagementSystem/Client/test/Test_FTS3Objects.py:175]: Before tests were never going through RSS for finding the status of a SE, now they do. So, some mocking of `StorageFactory` had to be added.

**Comment by fstagni** [comment]
OK now? I will merge in an hour or 2
